core/string/ustring.cpp

```diff
@@ -3083,59 +3083,28 @@ String String::substr(int p_from, int p_chars) const {
 
 int String::find(const String &p_str, int p_from) const {
 	if (p_from < 0) {
-		return -1;
-	}
-
-	const int src_len = p_str.length();
-
-	const int len = length();
-
-	if (src_len == 0 || len == 0) {
-		return -1; // won't find anything!
+		p_from = length() - p_str.length() + p_from + 1;
 	}
-
-	if (src_len == 1) {
-		return find_char(p_str[0], p_from); // Optimize with single-char find.
+	if (p_from < 0 || p_from > length() - p_str.length() || p_str.is_empty()) {
+		return -1; // Still out of bounds
 	}
 
-	const char32_t *src = get_data();
-	const char32_t *str = p_str.get_data();
-
-	for (int i = p_from; i <= (len - src_len); i++) {
-		bool found = true;
-		for (int j = 0; j < src_len; j++) {
-			int read_pos = i + j;
-
-			if (read_pos >= len) {
-				ERR_PRINT("read_pos>=len");
-				return -1;
-			}
-
-			if (src[read_pos] != str[j]) {
-				found = false;
-				break;
-			}
-		}
-
-		if (found) {
-			return i;
-		}
+	if (p_str.length() == 1) {
+		// Optimize with single-char implementation.
+		return span().find(p_str[0], p_from);
 	}
 
-	return -1;
+	return span().find_sequence(p_str.span(), p_from);
 }
 
 int String::find(const char *p_str, int p_from) const {
-	if (p_from < 0 || !p_str) {
-		return -1;
-	}
-
 	const int src_len = strlen(p_str);
 
-	const int len = length();
```

Comment:
I'm not 100% sure whether `length()` can be optimized out in `String`, since it's in RAM rather than on the stack. It may be better to just keep it in `const int len`, should also reduce your line change count again :)